### PR TITLE
docs/howto: document our release process

### PR DESCRIPTION
**Pull request by ttaylorr**
This pull request takes the long-overdue step of documenting our release process in a public fashion, so that others can release Git LFS, too.

##

/cc @git-lfs/core

### CODE REVIEW

File: docs/howto/release-git-lfs.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+`v2.n`-series. Let `v2.n-1` denote the previous release series.
+
+  1. First, we write the release notes and do the housekeeping required to
+     indicate a new version. On a new branch, called `release-next, do the
+     following:
+
+     * Run `script/changelog v2.n-1.0...HEAD` and categorize each merge commit
+       as a feature, bug-fix, miscellaneous change, or skipped. Ensure that your
+       `~/.netrc` credentails are kept up-to-date in order to make requests to
+       the GitHub API`.
```
Comment (bk2204) [inline docs/howto/release-git-lfs.md]: And it looks like that backtick has crept down to here.

File: docs/howto/release-git-lfs.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+       additional instructions in the section "For Windows" below.
+
+     * To build `*.deb` and `*.rpm` artifacts, follow the additional
+       instructions in "For Linux" below.
+
+  5. Create a GitHub release for the new version number, including the
+     changelog and the standard copy materials, as below:
+
+     ```
+     <CHANGEGLOG.md additions>
```
Comment (bk2204) [inline docs/howto/release-git-lfs.md]: It looks like there’s an extra “G” in “CHANGELOG”.

**Comment by ttaylorr** [comment]
@bk2204 Thanks, I fixed things up and am attaching a `range-diff`:

```
1:  f46ff56a ! 1:  2fdfe8d9 docs/howto: document our release process
    @@ -17,11 +17,11 @@
     +We follow Semantic Versioning standards as follows:
     +
     +  * `MAJOR` releases are done on a scale of 1-2 years. These encompass breaking,
    -+    incompabitble API changes, or command line interface changes that would
    ++    incompatible API changes, or command-line interface changes that would
     +    cause existing programs of use-cases of Git LFS to break.
     +
     +  * `MINOR` releases are done on a scale of 1-2 months. These encompass new
    -+    features, bug fixes, and other "medium"-sized changes into a semi-reguglar
    ++    features, bug fixes, and other "medium"-sized changes into a semi-regular
     +    release schedule.
     +
     +  * `PATCH` releases are done on the scale of 1-2 weeks. These encompass
    @@ -74,13 +74,13 @@
     +`v2.n`-series. Let `v2.n-1` denote the previous release series.
     +
     +  1. First, we write the release notes and do the housekeeping required to
    -+     indicate a new version. On a new branch, called `release-next, do the
    ++     indicate a new version. On a new branch, called `release-next`, do the
     +     following:
     +
     +     * Run `script/changelog v2.n-1.0...HEAD` and categorize each merge commit
     +       as a feature, bug-fix, miscellaneous change, or skipped. Ensure that your
     +       `~/.netrc` credentails are kept up-to-date in order to make requests to
    -+       the GitHub API`.
    ++       the GitHub API.
     +
     +       This will write a portion of the CHANGELOG to stdout, which you should
     +       copy and paste into `CHANGELOG.md`, along with an H2-level heading
    @@ -141,7 +141,7 @@
     +     changelog and the standard copy materials, as below:
     +
     +     ```
    -+     <CHANGEGLOG.md additions>
    ++     <CHANGELOG.md additions>
     +
     +     ## Packages
     +
```

@PastelMobileSuit: want to take a look before we merge this?

**Comment by ttaylorr** [comment]
@PastelMobileSuit thanks for your review, your comments were all very helpful! My latest push should take care of them.

File: docs/howto/release-git-lfs.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+  7. Push the tag, via:
+
+     ```ShellSession
+     $ git push origin v2.n.m
+     ```
+
+     And publish the release on GitHub.
+
+	8. Update the `_config.yml` file in
```
Comment (PastelMobileSuit) [inline docs/howto/release-git-lfs.md]: It looks like this step is improperly indented, and that's also causing it to be treated similarly to a code-block when the markdown is rendered.

**Pull request merged**